keras_nlp/samplers/beam_sampler.py

```diff
@@ -77,7 +79,13 @@ def __call__(
         index=0,
         mask=None,
         end_token_id=None,
+        return_all_beams=None,
```

Comment:
Do we need this in both places? I think we can probably just take this in at init time.

Is there a workflow that would need this at call time?

Reply:
Yes, in the `beam_sampler_test.py`:
```
def test_return_all_beams(self):
        state_chars = list("sequentially")
        state = tf.constant([[self.char_lookup[c] for c in state_chars]])
        prompt = tf.fill((self.batch_size, self.length), self.char_lookup["z"])
        output = self.sampler(
            next=self.next,
            prompt=prompt,
            state=state,
            return_all_beams=True,
        )
```
Here, inside the `self.sampler` method which uses the call. Not sure if I can initialize the `return_all_beams=True` somewhere else

Reply:
Fixed in the latest commit